kernel/src/engine/default/json.rs

```diff
@@ -88,6 +90,51 @@ impl<E: TaskExecutor> JsonHandler for DefaultJsonHandler<E> {
             self.readahead,
         )
     }
+
+    // note: for now we just buffer all the data and write it out all at once
+    fn write_json_file(
+        &self,
+        path: &url::Url,
+        data: Box<dyn Iterator<Item = Box<dyn EngineData>> + Send + '_>,
```

Comment:
Lifetime specification is since it defaults to `'static` and we don't want to impose that requirement. seemed odd to me that it would default to static but didn't take too much time to dive into it yet. and secondarily I think if we use `impl Iterator` it causes the trait not to be [object-safe](https://doc.rust-lang.org/reference/items/traits.html#object-safety)

Reply:
I'm seeing a compiler error that whenever we have `impl Iterator`: it breaks `JsonHandler` from being object-safe (which it must be since we use it as a trait object in `Arc<dyn JsonHandler>`

And you're right - didn't need the `'_` at all :)